R/zchunk_L102.GDP.R

```diff
@@ -42,121 +121,204 @@ module_socioeconomics_L102.GDP_DISABLED <- function(command, ...) {
+          arrange(year) %>% group_by(iso) %>%
+          mutate(gdp = cumprod(gdp.ratio)) %>% # actually ratio of gdp to base-year
+                                        # gdp, but we're calling it "gdp" so
+                                        # that join.gdp.ts() can work with it.
+          select(iso, year, gdp)
+
+    gdp.mil90usd.imf.country.yr <-
+        select(gdp_mil90usd_ctry, iso, year, gdp) %>% #strip region id col b/c
+                                        #imfgdp.ratio doesn't have one.
+        join.gdp.ts(imfgdp.ratio, 'iso')
+    ## columns: iso, year, gdp
+
+    ## Aggregate by GCAM region
+    gdp.mil90usd.imf.rgn.yr <-
+        left_join_error_no_match(gdp.mil90usd.imf.country.yr,
+                                 iso_region32_lookup, by= 'iso') %>%
+          group_by(GCAM_region_ID, year) %>%
+          summarise(gdp = sum(gdp)) %>%
+          filter(year %in% c(HISTORICAL_YEARS, FUTURE_YEARS))
+    ## columns:  GCAM_region_ID, year, gdp
+
+    ## join the IMF near future up to the SSP distant future, rename scenarios
+    ## to gSSP*
+    gdp.mil90usd.gSSP.rgn.yr <-
+        join.gdp.ts(gdp.mil90usd.imf.rgn.yr, gdp_bilusd_rgn_Yfut, 'GCAM_region_ID') %>%
+          mutate(scenario = paste0('g', scenario))
+    ## columns: scenario, GCAM_region_ID, year, gdp
+
+    ## combine SSP and gSSP scenarios into a single table (this will be one of
+    ## our final outputs)
+    gdp.mil90usd.scen.rgn.yr <-
+        bind_rows(gdp.mil90usd.SSP.rgn.yr, gdp.mil90usd.gSSP.rgn.yr)
+
+    ## Construct a table of population by scenario, region, and year.  We have a
+    ## table of historical population, and a table of future population by
+    ## scenario, both in wide form.  Convert to long form and filter to the years
+    ## we need.  Add a scenario column to historical years, and combine the
+    ## whole thing into a single table.
+    pop.thous.fut <-
+        rename(L101.Pop_thous_Scen_R_Yfut, population = value) %>%
+          filter(year %in% FUTURE_YEARS)
+    pop.thous.hist <-
+        rename(L101.Pop_thous_R_Yh, population = value) %>%
+          filter(year %in% HISTORICAL_YEARS) %>%
+          tidyr::crossing(scenario = unique(pop.thous.fut[['scenario']]))
+    pop.thous.scen.rgn.yr <-
+        bind_rows(pop.thous.hist, pop.thous.fut) %>%
+          mutate(year = as.integer(year), population=as.numeric(population)) %>%
+          select(scenario, GCAM_region_ID, year, population)
+
+    ## calculate per-capita GDP.  This is another final output
+    pcgdp.thous90usd.scen.rgn.yr <-
+        left_join(gdp.mil90usd.scen.rgn.yr, pop.thous.scen.rgn.yr,
+                  by = c('scenario', 'GCAM_region_ID', 'year')) %>%
+          mutate(pcgdp = gdp / population) %>%
+          select(scenario, GCAM_region_ID, year, pcgdp)
+
+    ## Calculate the PPP-MER conversion factor for each region.  As far as I can
+    ## tell, we do this only in 2010 and assume it to be constant for all time
```

Comment:
No, I think this is working as designed, and it's not obvious (to me at least) what you would do instead.  It just confused me a bit when I was first working through it.  Haewon updated the comment a bit to clarify that these assumptions are by design.